controllers/users_controller.js

```diff
@@ -0,0 +1,34 @@
+const User = require('../models/users_model');
+
+const getAllUsers = async (req, res) => {
+    try {
+      const users = await User.find();
+      res.json(users);
+    } catch (err) {
+        console.warn("Error fetching users:", err);
+        res.status(500).json({ error: "An error occurred while fetching the users." });
+      }
+  };
+  
+const registerNewUser = async (req, res) => {
+  try {
+    const {username, email, password} = req.body;
+    const user = new User({
+      username,
+      email,
+      password
+    });
+
+    const savedUser = await user.save();
+    res.json(savedUser);
+  } catch (err) {
+    console.warn("Error registering user:", err);
+    if (err.code === 11000) {
+      res.status(500).json({ error: "username already exsits."});
```

Comment:
```suggestion
      res.status(400).json({ error: "username already exsits."});
```
This is a client error, not a server error